src/Admin/Pool.php

```diff
@@ -235,20 +235,21 @@ public function hasAdminByClass($class)
      *
      * @throws \InvalidArgumentException if the root admin code is an empty string
      *
-     * @return \Sonata\AdminBundle\Admin\AdminInterface|false
+     * @return AdminInterface|false
      */
     public function getAdminByAdminCode($adminCode)
     {
         if (!\is_string($adminCode)) {
             @trigger_error(sprintf(
-                'Passing a non string value as argument 1 for %s() is deprecated since sonata-project/admin-bundle 3.x and will throw an exception in 4.0.',
+                'Passing a non string value as argument 1 for %s() is deprecated since sonata-project/admin-bundle 3.x and will result in a PHP TypeError in 4.0.',
```

Comment:
IMO, if you want to be specific on the exception type, the message should be:
```suggestion
                'Passing a non string value as argument 1 for %s() is deprecated since sonata-project/admin-bundle 3.x and will throw `\TypeError` exception in 4.0.',
```
There is no reason for me to mention "PHP" in the message.

Reply:
Changed it to "will cause a \TypeError". It won't be thrown

Reply:
Just for the records, from the PHP documentation: https://www.php.net/manual/en/class.typeerror.php#typeerror.intro
>There are three scenarios where a TypeError may be thrown.

Reply:
Thank you @phansys for pointing it out. I am aware of this. Only the first scenario applies here